### DISCUSSION TITLE
ImageOptimizer plugin not working

### DISCUSSION THREAD

**Discussion by larsjanssen6** [statamic/cms#6773] [Troubleshooting] [answered]
Hi, 

We bought the [ImageOptimizer](https://statamic.com/addons/4rn0/imageoptimizer-v3) plugin. It seems to be configured correctly:

![Screenshot 2022-09-23 at 11 20 34](https://user-images.githubusercontent.com/7254997/191930262-c96af97b-81ed-4bda-8e10-c2e1a6b0caaa.png)

However, when I upload an image (assets), this error is given:

```
[previous exception] [object] (League\\Flysystem\\FileExistsException(code: 0): File already exists at path: 9209455c-5464-466f-a32c-7cf6b0ba2f04-1663923608.jpeg at /home/forge/domain.nl/vendor/league/flysystem/src/Filesystem.php:406)
[stacktrace]
```

The error is given for every image I upload. We are using s3 for storing our resources. When I disable the plugin uploading is working. 

What could be wrong here?

**Comment by duncanmcclean** [comment, marked as answer]
Please get in touch directly with the developer of the addon. 

They don't seem to have GitHub Issues enabled but looks like you can contact them via email on [their Marketplace page](https://statamic.com/addons/4rn0/imageoptimizer-v3) (see the 'Get Support' button).

**Comment by larsjanssen6** [comment]
Allright thankyou
